packages/workflow/src/workflow.ts

```diff
@@ -538,29 +587,45 @@ export type ActivityInterfaceFor<T> = {
  *
  * export function execute(): Promise<void> {
  *   const response = await httpGet("http://example.com");
+ *   // Or with custom options:
+ *   const response2 = await httpGetFromJava.runWithOptions({
+ *     staticSummary: 'Java HTTP call with timeout override',
+ *     startToCloseTimeout: '2m'
+ *   }, ["http://fast-api.example.com"]);
  *   // ...
  * }
  * ```
  */
-export function proxyActivities<A = UntypedActivities>(options: ActivityOptions): ActivityInterfaceFor<A> {
+export function proxyActivities<A = UntypedActivities>(options: ActivityOptions): ActivityInterfaceWithOptions<A> {
   if (options === undefined) {
     throw new TypeError('options must be defined');
   }
   // Validate as early as possible for immediate user feedback
   validateActivityOptions(options);
-  return new Proxy(
-    {},
-    {
+
+  function createActivityProxy(options: ActivityOptions): ActivityInterfaceWithOptions<A> {
+    return new Proxy({} as ActivityInterfaceWithOptions<A>, {
       get(_, activityType) {
         if (typeof activityType !== 'string') {
           throw new TypeError(`Only strings are supported for Activity types, got: ${String(activityType)}`);
         }
-        return function activityProxyFunction(...args: unknown[]): Promise<unknown> {
-          return scheduleActivity(activityType, args, options);
+
+        function activityProxyFunction(...args: unknown[]): Promise<unknown> {
+          return scheduleActivity(activityType as string, args, options);
+        }
+
+        activityProxyFunction.runWithOptions = function (
+          overrideOptions: ActivityOptions,
+          args: any[]
+        ): Promise<unknown> {
+          return scheduleActivity(activityType, args, { ...options, ...overrideOptions });
```

Comment:
That won't be so easy. You will need to merge recursively on retry policy and priority.

Reply:
added a `deepMerge` utility in `object-helpers.ts` for use here (and possibly elsewhere in the future)